### PR TITLE
change api connector details from inline editor

### PR DESCRIPTION
**Pull request by michael-coker**
fix https://github.com/syndesisio/syndesis/issues/2934

one thing I couldn't figure out was how to reset the form to the default/original values if you edit the form and click "cancel"

### CODE REVIEW

File: app/ui/src/app/customizations/api-connector/api-connector-info/api-connector-info.component.ts [hunk 1]
```diff
@@ -52,63 +60,77 @@ export class ApiConnectorInfoComponent implements OnInit {
...
 
-  @HostListener('document:click', ['$event'])
-  private onDocumentClick(event: Event): void {
-    const element: any = event.target;
-    const elementTag = element.tagName.toLowerCase();
-    if (elementTag !== 'input' && elementTag !== 'textarea' && this.editControlKey) {
-      setTimeout(() => this.onSubmit(), 0);
+  onSubmit(): void {
+    if (this.apiConnectorDataForm.valid) {
+      if (this.isDirty) {
```
Comment (gashcrumb) [inline app/ui/src/app/customizations/api-connector/api-connector-info/api-connector-info.component.ts]: `isDirty` can probably be replaced with looking at `this.apiConnectorDataForm.dirty`

File: app/ui/src/app/customizations/api-connector/api-connector-info/api-connector-info.component.ts [hunk 2]
```diff
@@ -52,63 +60,77 @@ export class ApiConnectorInfoComponent implements OnInit {
...
+
+  clearValue(el) {
+    el.nativeElement.value = '';
+  }
+
+  isDisabled() {
+    return !this.editMode;
+  }
+
+  onChange() {
```
Comment (gashcrumb) [inline app/ui/src/app/customizations/api-connector/api-connector-info/api-connector-info.component.ts]: Check if there's a `dirty` property on `apiConnectorDataForm`, if there is, then you don't need this.

File: app/ui/src/app/customizations/api-connector/api-connector-info/api-connector-info.component.ts [hunk 3]
```diff
@@ -52,63 +60,77 @@ export class ApiConnectorInfoComponent implements OnInit {
...
+    this.enableEdit();
   }
 
-  onEditEnable(event: Event, key: string): void {
-    event.preventDefault();
-    event.stopPropagation();
-    this.editControlKey = key;
+  enableEdit() {
+    this.editMode = true;
+    this.apiConnectorDataForm.enable();
```
Comment (gashcrumb) [inline app/ui/src/app/customizations/api-connector/api-connector-info/api-connector-info.component.ts]: Can probably grab the values from `apiConnectorDataForm` here like:

`this.oldValues = { ...apiConnectorDataForm.value };`

Then when the user clicks 'Cancel':

`this.apiConnectorDataForm.value = this.oldValues`

which I think *should* reset the values displayed in the form.

File: app/ui/src/app/customizations/api-connector/api-connector-info/api-connector-info.component.ts (hunk 2 above)
Comment (michael-coker) [reply app/ui/src/app/customizations/api-connector/api-connector-info/api-connector-info.component.ts]: `onChange()` only handles the `input type="file"` used to upload an image, and sets an `iconFile` var to populate the thumbnail `img`. I'm not sure how `dirty` could be used here?

File: app/ui/src/app/customizations/api-connector/api-connector-info/api-connector-info.component.ts (hunk 3 above)
Comment (michael-coker) [reply app/ui/src/app/customizations/api-connector/api-connector-info/api-connector-info.component.ts]: looks like `value` is a read only property, so I can't overwrite it, but I can use `setValue()`.

**Comment by michael-coker** [comment]
@gashcrumb thanks for the feedback! updated with these changes.

**Pull request merged**